aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+
+### AngularJS
+
+AngularJS uses a convention to address CSRF. If the server sends a cookie with the name ``XSRF-TOKEN``, the Angular ``$http`` service will add the value from this cookie to a header when it sends a request to this server. This process is automatic; you don't need to set the header explicitly. The header name is ``X-XSRF-TOKEN``. The server should detect this header and validate its contents. To have your ASP.NET Core API work with this convention, you should configure your app to provide a token in a cookied called ``XSRF-TOKEN``:
+
+Also, configure the antiforgery service to look for a header named ``X-XSRF-TOKEN``:
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddAntiforgery(options => options.HeaderName = "X-XSRF-TOKEN");
+```
+
+### JavaScript
+
+The implementation for non-AngularJS JavaScript is very similar. As above, you should generate the token on the server and send it to the client in a cookie:
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+context.Response.Cookies.Append("CSRF-TOKEN", tokens.RequestToken, 
+  new Microsoft.AspNetCore.Http.CookieOptions { HttpOnly = false });
+```
+
+Then, assuming you construct your script requests to send the token in a header called ``X-CSRF-TOKEN``, configure the antiforgery service to look for this header:
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddAntiforgery(options => options.HeaderName = "X-CSRF-TOKEN");
+```
+
+An example of making an AJAX request with the appropriate header, from jQuery:
+<!-- literal_block {"xml:space": "preserve", "language": "javascript", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```javascript
+var csrfToken = $.cookie("CSRF-TOKEN");
+
+$.ajax({
+    url: "/api/password/changepassword",
+    contentType: "application/json",
+    data: JSON.stringify({ "newPassword": "ReallySecurePassword999$$$" }),
+    type: "POST",
+    headers: {
+        "X-CSRF-TOKEN": csrfToken
+    }
+});
+```
+
+## Using IAntiforgery
+
+`IAntiforgery` provides access to the antiforgery system and can be requested in the `Configure` method of the `Startup` class. The following example uses middleware from the app's home page to generate an antiforger token and send it in the response as a cookie (using the default Angular naming convention described above):
```

Comment:
`IAntiforgery` provides access to the antiforgery system.  It can be requested in the `Configure` method of the `Startup` class. 

Please find out the preferred spelling. I think it might be anti-forgery but I can't find anything definitive.